Source/NSubstitute/SubstituteExtensions.cs

```diff
@@ -142,7 +142,7 @@ internal static ConfiguredCall Returns<T>(MatchArgs matchArgs, T returnThis, par
             }
             else
             {
-                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese));
+                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese).ToArray());
```

Comment:
Please pass an `IEnumerable` here and later make array in constructor if you need that. It's implementation detail that IEnumerables should be converted to arrays.

Moreover, we don't even need that - just initialize `ConcurrentQueue` with `IEnumerable` and later call `LastOrDefault()` on the created queue.